lib/build_cache.py

```diff
@@ -804,9 +839,18 @@ def find_commit(self, git_id):
       # collisions.
       cp = self.git(["log", "--format=%h%n%B", "-n", "1", git_id],
                     fail_ok=True)
+      deleted = self.git(["log", "--format=%h%n%B", "-n", "1",
+                          "&%s" % git_id],fail_ok=True)
```

Comment:
`find_commit()` gets used a lot and previously had no side effects, IIRC. I'm skeptical a new side effect should be rescuing deleted branches or making anything ready.

Is what you're after to remove tags that no longer point to deleted branches? If so I wonder if that's more appropriate in `ready()`.